src/centroidal_impedance_controller.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file
+ * @license BSD 3-clause
+ * @copyright Copyright (c) 2021, New York University and Max Planck
+ * Gesellschaft
+ *
+ * @brief Implements the CentroidalImpedance controller.
+ */
+
+#include "mim_control/centroidal_impedance_controller.hpp"
+
+#include "pinocchio/algorithm/frames.hpp"
+
+namespace mim_control
+{
+CentroidalImpedanceController::CentroidalImpedanceController()
+{
+}
+
+void CentroidalImpedanceController::initialize(
+    const double& mass,
+    Eigen::Ref<const Eigen::Vector3d> inertia,
+    const pinocchio::Model& pinocchio_model,
+    const std::string& root_frame_name,
+    const std::vector<std::string>& end_frame_names,
+    double friction_coeff,
+    Eigen::Ref<const Vector6d> qp_penalty_weights,
+    Eigen::Ref<const Eigen::Vector3d> kc,
+    Eigen::Ref<const Eigen::Vector3d> dc,
+    Eigen::Ref<const Eigen::Vector3d> kb,
+    Eigen::Ref<const Eigen::Vector3d> db,
+    Eigen::Ref<const Array6d> frame_placement_error_gain,
+    Eigen::Ref<const Array6d> frame_velocity_error_gain
+)
+{
+    bool is_free_flyer =
+        pinocchio_model.joints[1].shortname() == "JointModelFreeFlyer";
```

Comment:
Shouldn't it be `pinocchio_model.joints[0].shortname()` with a **0**?

Reply:
I copied the code from the code that was in place before. I think the index of zero is correct. Here is the output for solo12:

```python
$ pin_robot.model
Nb joints = 14 (nq=19,nv=18)
  Joint 0 universe: parent=0
  Joint 1 root_joint: parent=0
  Joint 2 FL_HAA: parent=1
  Joint 3 FL_HFE: parent=2
  Joint 4 FL_KFE: parent=3
  Joint 5 FR_HAA: parent=1
  Joint 6 FR_HFE: parent=5
  Joint 7 FR_KFE: parent=6
  Joint 8 HL_HAA: parent=1
  Joint 9 HL_HFE: parent=8
  Joint 10 HL_KFE: parent=9
  Joint 11 HR_HAA: parent=1
  Joint 12 HR_HFE: parent=11
  Joint 13 HR_KFE: parent=12
```

Note how the zero joint is the universe. The first joint is the free-flying joint:

```
$ robot.pin_robot.model.joints[1]
JointModelFreeFlyer
  index: 1
  index q: 0
  index v: 0
  nq: 7
  nv: 6
```

I assume things are the same in C++ as they are in python.